framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -696,7 +691,13 @@ AutomaticMortarGeneration::buildMortarSegmentMesh()
         std::abs(msinfo.xi2_b) > 1.0 + TOLERANCE)
     {
       // Erase from secondary to msms map
-      libmesh_map_find(_secondary_elems_to_mortar_segments, msinfo.secondary_elem).erase(msm_elem);
+      auto it = _secondary_elems_to_mortar_segments.find(msinfo.secondary_elem);
+      mooseAssert(it != _secondary_elems_to_mortar_segments.end(),
+                  "We should have found the element");
+      auto & msm_set = it->second;
+      msm_set.erase(msm_elem);
+      if (msm_set.empty())
+        _secondary_elems_to_mortar_segments.erase(it);
```

Comment:
Actually I think this would be the most appropriate for the nodal normal comment, since it is the nodes with secondary element neighbors (not mortar segment mesh neighbors) that determine the nodal normal field